### ISSUE TITLE
Stopping the speaker resets the playlist to the first entry

### ISSUE THREAD

**Issue by d34vbd5** [nfarina/homebridge-sonos#22]
Hi, 
first: thank you, great plugin!
I got one little issue, if i stop a speaker using "turn off 'speakername'" it resets the current Sonos playlist to the first entry. As soon as i use "turn on 'speakername'" it does not continue playing but starts all over again.

Any way to fix this?

**Comment by thezepter** [comment]
Confirmed... same Issue oder Feature here.

**Comment by edelmaca** [comment]
I happened to run in this problem too. I hope someone can figure this out and make it work like we press the pause button.

**Comment by BrekiTomasson** [comment]
Yeah, this seems like a pretty dumb bug. Other Sonos APIs use the "pause" command rather than simply stopping all playback.

**Comment by BrekiTomasson** [comment]
Ping @edelmaca @thezepter and @d34vbd5 - I have a pull request with a fix here now. Just waiting for the Powers That Be to approve it and implement it in the master build. Until then, you can rewrite the line that has `this.device.stop` in it to `this.device.pause` to fix it on your end.

**Comment by d34vbd5** [comment]
Thanks!

What's the path to the file where the change has to be done? I can't find my homebridge plugins path...

**Comment by thezepter** [comment]
i think.. 

/usr/local/lib/node_modules

**Comment by edelmaca** [comment]
@BrekiTomasson These changes seem to solve the problem. Perfect. Thank you for that.

**Comment by d34vbd5** [comment]
Works like a charm!
Thank you

**Comment by Nousemusic** [comment]
Works for me. Thank you very much.

**Comment by BrekiTomasson** [comment]
Glad to hear everybody liking the fix! Anyway, solved in #24 if we can get it approved. :)

**Comment by gizbod** [comment]
I have rewritten the line as suggested above but the problem seems to of not been rectified? what am i doing wrong?

**Comment by d34vbd5** [comment]
I would guess you are doing it wrong.
Works fine here

**Comment by gizbod** [comment]
Yea I would guess so. I wonder if I changed the right thing, pretty sure I did though :s

**Comment by gizbod** [comment]
Even if i delete all the code from the index.js file the plugin still works and I can control the on off of the speakers through siri. so obviously whatever edit i make is having no affect whatsoever. I'm confused with this.

**Comment by d34vbd5** [comment]
You do have to restart homebridge

**Comment by gizbod** [comment]
Yep, I restarted homebridge also.

**Comment by gizbod** [comment]
thought i would update that i now have this working. I had home bridge set up on my macbook and it wouldn't work but now set up on RPi editing the line has solved the problem.

**Comment by nmyster** [comment]
Damn, I had this fixed a couple of months ago but never got around to PR'ing it - glad its coming though
